### PR TITLE
Pys 24

### PR DESCRIPTION
**Pull request by badaniya**
Exposed ca_cert and rest_proto parameters out to the pyswitchlib asset and pyswitch device for now.   Need to discuss with team on how we should really handle exposing these parameters out.

This is just a POC for https for now.

### CODE REVIEW

File: pyswitchlib/asset.py [hunk 1]
```diff
@@ -38,6 +38,7 @@ def on_deletion (killed_ref):
 
         self._ip_addr = ip_addr
         self._auth = auth
+        self._rest_protocol = 'http'
```
Comment (samirsss) [inline pyswitchlib/asset.py]: Can we default to https?

File: pyswitchlib/asset.py [hunk 2]
```diff
@@ -49,6 +50,19 @@ def on_deletion (killed_ref):
...
 
+        if rest_proto:
+            if rest_proto.lower() == 'http' or rest_proto.lower() == 'https':
+                self._rest_protocol = rest_proto.lower()
+            else:
+                raise RestProtocolTypeError("Rest protocol type must be 'http' or 'https'.  '" + rest_proto + "' was specified.")
+
+        if ca_cert:
+            if os.path.isfile(ca_cert):
+                self._session.verify = ca_cert
```
Comment (samirsss) [inline pyswitchlib/asset.py]: Are there any other verifications that can be done except for file exists? I dont know and hence asking

File: pyswitch/XMLAsset.py [hunk 3]
```diff
@@ -5,8 +5,10 @@
 
 
 class XMLAsset(Asset):
-    def __init__(self, ip_addr='', auth=('admin', 'password'), fw_ver='', timeout=''):
-        super(XMLAsset, self).__init__(ip_addr=ip_addr, auth=auth, fw_ver=fw_ver, timeout=timeout)
+    def __init__(self, ip_addr='', auth=('admin', 'password'),
+                 rest_proto='', ca_cert='', fw_ver='', timeout=''):
```
Comment (samirsss) [inline pyswitch/XMLAsset.py]: Can we default to https for rest_proto here?

Comment (badaniya) [reply pyswitch/XMLAsset.py]: Synced up with Sunil C. and HTTPS is not enabled on the switch (nos/slxos) by default.  CA cert needs to be added and server cert enrolled.  HTTPS and HTTP are mutually exclusive except there maybe a special NOS build where both HTTP/HTTPS can be enabled at the same time.  If we have HTTPS by default then asset construction/connection would typically take a hit trying HTTPS first, then HTTP.

**Pull request merged**